tests/TestCase/Middleware/UnauthorizedHandler/RedirectHandlerTest.php

```diff
@@ -160,4 +160,48 @@ public function testHandleException()
         $this->expectException(Exception::class);
         $handler->handle($exception, $request);
     }
+
+    public function testHandleRedirectionWithExtension(): void
+    {
+        $handler = new RedirectHandler();
+
+        $exception = new Exception();
+        $request = ServerRequestFactory::fromGlobals(
+            ['REQUEST_METHOD' => 'GET'],
+        );
+        $request = $request->withParam('_ext', 'csv');
+
+        $this->expectException(Exception::class);
+
+        $handler->handle($exception, $request, [
+            'exceptions' => [
+                Exception::class,
```

Comment:
```suggestion
        $exception = new LogicException();
        $request = ServerRequestFactory::fromGlobals(
            ['REQUEST_METHOD' => 'GET'],
        );
        $request = $request->withParam('_ext', 'csv');

        $this->expectException(LogicException::class);

        $handler->handle($exception, $request, [
            'exceptions' => [
                LogicException::class,
```

Any concern on this being a broad type expectation? You could use something more narrow `LogicException` and have more tightly scoped behavior captured in the test.